### PR TITLE
Feature/i os 152

### PR DESCRIPTION
**Pull request by gzphreak**
Add the ability for Kodi to start it's web services on an automatically selected, by the OS, port.

### CODE REVIEW

File: system/settings/settings.xml [hunk 1]
```diff
@@ -1712,9 +1712,9 @@
         </setting>
         <setting id="services.webserverport" type="integer" parent="services.webserver" label="730" help="36329">
           <level>1</level>
-          <default>8080</default>
+          <default>0</default>
           <constraints>
-            <minimum>1</minimum>
+            <minimum>0</minimum>
```
Comment (jjoyce-bittorrent) [inline system/settings/settings.xml:1717]: what about using public port range?

File: xbmc/network/NetworkServices.cpp [hunk 2]
```diff
@@ -1060,7 +1060,7 @@ bool CNetworkServices::StopZeroconf()
 
 bool CNetworkServices::ValidatePort(int port)
 {
-  if (port <= 0 || port > 65535)
+  if (port < 0 || port > 65535)
```
Comment (jjoyce-bittorrent) [inline xbmc/network/NetworkServices.cpp:1063]: should this be in public port range too?

**Review by jjoyce-bittorrent — APPROVED** [review]
small comments. lgtm.

File: system/settings/settings.xml (hunk 1 above)
Comment (gzphreak) [reply system/settings/settings.xml:1717]: In this case 0 lets the library know to to use a random port via the OS.

File: xbmc/network/NetworkServices.cpp (hunk 2 above)
Comment (gzphreak) [reply xbmc/network/NetworkServices.cpp:1063]: Maybe? I didn't think to limit it to that range as Kodi was already 1 to max.

**Pull request merged**